Source/SteveSloth/Private/MyEnemyAttackingState.cpp

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************************
+ * Copyright: SteveSloth
+ * Name: Ken Ferris
+ * Script: MyEnemyAttackingState.cpp
+ * Date: May 3, 2024
+ * Description: Enemy Attack State
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+
+#include "MyEnemyAttackingState.h"
+#include "MyEnemyBaseClass.h"
+#include "Kismet/GameplayStatics.h"
+
+void UMyEnemyAttackingState::EnterState()
+{
+
+}
+
+void UMyEnemyAttackingState::ExitState()
+{
+
+}
+
+void UMyEnemyAttackingState::UpdateState(float deltaTime)
+{
+	if (IsPlayerInRange())
+	{
+		AttackPlayer();
+	}
+}
+
+bool UMyEnemyAttackingState::IsPlayerInRange() const
```

Comment:
Dont need to check range that is/will be done in the state machine. This script should be what happens when the enemy is attacking. I would remove the whole function. Just wondering as to why this function is const?